tests/isolated/patcher_existing_locks_preexisting.py

```diff
@@ -0,0 +1,41 @@
+"""
+Ensure pre-existing RLocks get upgraded in a variety of situations.
+"""
+
+import sys
+import threading
+import unittest.mock
+import eventlet
+
+python_lock = threading._PyRLock
+
+
+class NS:
+    lock = threading.RLock()
+
+    class NS2:
+        lock = threading.RLock()
+
+    dict = {1: 2, 12: threading.RLock()}
+    list = [0, threading.RLock()]
+
+
+def ensure_upgraded(lock):
+    if not isinstance(lock, python_lock):
+        raise RuntimeError(lock)
+
+
+if __name__ == '__main__':
+    # These extra print()s caused either test failures or segfaults until
+    # https://github.com/eventlet/eventlet/issues/864 was fixed.
+    if sys.version_info[:2] > (3, 9):
+        print(unittest.mock.NonCallableMock._lock)
+    print(NS.lock)
```

Comment:
I'm confused about how this passes with these prints...

Oh, https://github.com/eventlet/eventlet/commit/e51f72a60c868b7b88e2c5758b5c65d6f0ab446d started allowing multiple lines as long as the last one was `pass`... why'd we do that?

Oh. Hmm...

```
Program .../tests/isolated/patcher_fork_after_monkey_patch.py output:
---
.../tests/isolated/patcher_fork_after_monkey_patch.py:38: DeprecationWarning: This process (pid=166324) is multi-threaded, use of fork() may lead to deadlocks in the child.
  if os.fork() == 0:
pass

---
```

Reply:
Yeah, I made those changes because of noise like that.